.gitmodules

```diff
@@ -4,3 +4,6 @@
 [submodule "uthash"]
 	path = uthash
 	url = https://github.com/troydhanson/uthash.git
+[submodule "uncrustify"]
```

Comment:
I would rather have fewer submodules than more. But apart from that, pulling this in this way would include it in the distributed source tarballs, which isn't good. Also, it's GPL. So we should really keep it at arm's length. Better just install it as a normal build dependency.

Reply:
The main problem with installing a system uncrustify version through `apt`/`yum`/`brew` is that it is important that everyone uses the exact same version. A newer version might have a formatting bug fixed and result in slightly different formatting. So a submodule seemed like a reasonable way to ensure a consistent version. (for black this is less of an issue, because we can just say we should use the latest version, which is very easy to install using `pip`)

>  But apart from that, pulling this in this way would include it in the distributed source tarballs, which isn't good. 

You mean the ones created by `make dist`? Since it's not in the `EXTRA_DIST` list they won't be included there. Also github doesn't seem to include submodule contents in the source tarballs it generates.

> Also, it's GPL. So we should really keep it at arm's length. Better just install it as a normal build dependency.

Since we're not linking against it (only executing the built binary) I personally don't see much of an issue with it. 

To resolve concerns though, I now pushed a version where we get the tarball from github and compile the `uncrustify` binary in a temporary directory.

Reply:
We can just say, we use uncrustify version x.y.z, and everyone just has to get that one.

I hate git submodules and I don't want more git submodules. :)

Reply:
Did you see I updated the the PR to remove the submodule and have a make target to fetch and compile the uncrustify version from sources? Is that okay? Or would you rather I remove the install steps from the `Makefile` completely and instead move them to a `CONTRIBUTING.md` and have people run them manually?